backend/layers/business/business.py

```diff
@@ -340,6 +413,27 @@ def update_collection_version(
             self.database_provider.save_collection_publisher_metadata(version_id, publisher_metadata_to_set)
         self.database_provider.save_collection_metadata(version_id, new_metadata)
 
+        if all(
+            [not ignore_doi_update, new_doi != current_doi, FeatureFlagService.is_enabled(FeatureFlagValues.SCHEMA_4)]
+        ):
+            for dataset in current_collection_version.datasets:
+                if dataset.status.processing_status != DatasetProcessingStatus.SUCCESS:
+                    self.logger.info(
+                        f"Dataset {dataset.version_id.id} is not successfully processed. Skipping metadata update."
+                    )
+                    continue
```

Comment:
@nayib-jose-gloria this silent failure to update the doi here makes me antsy. I realize the updated Collection metadata is returned to the user, so they technically can verify that the metadata look as they wish, but the doi will be hidden amongst other features (and the others will have been updated!). Especially when using the Discover API...

Reply:
I should update this to read if processing_status == FAILURE

if a dataset makes it to this loop, it has either status SUCCESS or FAILURE, and if it's FAILURE, then its current metadata state is irrelevant since it requires a reupload to fix anyway. Unless there's an edge case I'm not considering?